### PR TITLE
Fix proxy tests

### PR DESCRIPTION
**Pull request by justinfagnani**
The probably was port contention between the proxied test server and polyserve. I also made teardown wait for servers to close.

### CODE REVIEW

File: src/test/start_server_test.ts [hunk 1]
```diff
@@ -97,11 +98,20 @@ suite('startServer', () => {
...
+      function _done() {
+        if (--waitingCount === 0) {
+          done();
+        }
+      }
+      if (proxyServer) {
+        waitingCount++;
+        proxyServer.close(_done);
+      }
+      app.close(_done);
```
Comment (rictic) [inline src/test/start_server_test.ts]: I think we should strongly prefer async/await over callback counting for all the usual reasons.

Comment (justinfagnani) [inline src/test/start_server_test.ts]: to use async await I have to convert to promises first

**Comment by TimvdLippe** [comment]
I was not aware of any failures, neither locally nor on Travis. Did I miss something? Thanks for fixing anyways!

**Comment by rictic** [comment]
Yeah, it's kinda strange. It never failed on travis, but both Justin and I were seeing failures locally.

¯\_(ツ)_/¯

**Pull request merged**